### PR TITLE
LDAP password policy

### PR DESCRIPTION
**Pull request by ZiBiS**
The password driver to  appropriate change password when  the PPolicy overlay is installed in openLDAP server.
In plugins/password/drivers/ldap.php i've made small change to work properly with  Net_LDAP2 in 2.2.0 version.

### CODE REVIEW

File: plugins/password/drivers/ldap_ppolicy.php [hunk 1]
```diff
@@ -0,0 +1,93 @@
...
+        	$result = stream_get_contents($pipes[1]);
+        	fclose($pipes[1]);
+        	
+        	$this->_debug('Result:'.$result);
+        	
+        	switch ($result) {
+        		case "OK":
+        			return PASSWORD_SUCCESS;
+        		case "Password is in history of old passwords":
+        			return  PASSWORD_IN_HISTORY;
```
Comment (alecpl) [inline plugins/password/drivers/ldap_ppolicy.php]: PASSWORD_IN_HISTORY constant does not exist. Did you forget to commit password.php changes?

Comment (ZiBiS) [inline plugins/password/drivers/ldap_ppolicy.php]: Yes, you're right. I've made new commit with it.

File: plugins/password/drivers/ldap_ppolicy.php [hunk 2]
```diff
@@ -0,0 +1,93 @@
...
+        $bindPW = $rcmail->config->get('password_ldap_searchPW');
+        $cafile = $rcmail->config->get('password_ldap_cafile');
+
+        $descriptorspec = array(
+        		0 => array("pipe", "r"),  // stdin is a pipe that the child will read from
+        		1 => array("pipe", "w"),  // stdout is a pipe that the child will write to
+        		2 => array("file", "/tmp/ppolicy_ldap.err", "a") // stderr is a file to write to
+        );
+        
+        $cwd = 'plugins/password/drivers/';
```
Comment (alecpl) [inline plugins/password/drivers/ldap_ppolicy.php]: Shouldn't this be a path to helpers directory? Also does the command need to be configurable at all?

Comment (ZiBiS) [inline plugins/password/drivers/ldap_ppolicy.php]: I changed it to logs directory. This can be useful in case of any errors.

Comment (alecpl) [inline plugins/password/drivers/ldap_ppolicy.php]: First of all you should use log_dir option to get the logs dir location. Second, I was talking about 'plugins/password/drivers/'.

Comment (ZiBiS) [inline plugins/password/drivers/ldap_ppolicy.php]: I'm sorry, I missed that path is not correct and log_dir should be get in propper way now.

**Comment by alecpl** [comment]
Last thing, all configuration options of this new driver must have prefix "password_ldap_ppolicy_". Please also solve the merge conflict.

**Pull request merged**